### PR TITLE
Bump phantompeakqualtools 1.2.2

### PR DESCRIPTION
**Pull request by JoseEspinosa**
Trying to update to the last available image of `phantompeaktools` for fixing [this](https://github.com/nf-core/chipseq/actions/runs/9680916237/job/26710242161?pr=391) error. 
Also adding `stub` and `nf-test` to the module.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`

### CODE REVIEW

**Comment by JoseEspinosa** [comment]
Since the last updated version of the image did not fix the docker issue above, I used the wave version of the containers instead.

File: modules/nf-core/phantompeakqualtools/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    tag "phantompeakqualtools"
+
+    test("sarscov2 - bam - single_end") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
```
Comment (SPPearce) [inline modules/nf-core/phantompeakqualtools/tests/main.nf.test:19]: This is a paired-end bam file.

File: modules/nf-core/phantompeakqualtools/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.spp[0][1]).match() },
+                { assert snapshot(
+                    process.out.uppercase.collect { it.collect { it instanceof String && file(it).exists() ? file(it).name : it }}
```
Comment (SPPearce) [inline modules/nf-core/phantompeakqualtools/tests/main.nf.test:30]: I don't know what this is trying to do.

**Review by SPPearce — COMMENTED** [review]
I'm going to simplify the tests, just give me a minute.

File: modules/nf-core/phantompeakqualtools/tests/main.nf.test (hunk 2 above)
Comment (JoseEspinosa) [reply modules/nf-core/phantompeakqualtools/tests/main.nf.test:30]: I took it from [here](https://github.com/nvnieuwk/nf-test-snippets/tree/main?tab=readme-ov-file) "Output everything from the process/workflow without checking md5sum of the files - only check existence"

**Review by SPPearce — APPROVED** [review]
I'm happy assuming the tests pass.

File: modules/nf-core/phantompeakqualtools/tests/main.nf.test [hunk 3]
```diff
@@ -70,7 +72,7 @@ nextflow_process {
                 """
                 input[0] = [
                     [ id:'test', single_end:false ], // meta map
-                    file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
```
Comment (JoseEspinosa) [inline modules/nf-core/phantompeakqualtools/tests/main.nf.test:75]: true 🤦‍♂️

**Comment by JoseEspinosa** [comment]
> I'm happy assuming the tests pass.

Thanks @SPPearce

**Pull request merged**